modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,188 @@
+          a set of LDAP queries under a specific category. The custom query is controlled via
+          the LDAPQUERY parameter, which will be used when the ACTION value is set to CUSTOM_QUERY.
+
+          Alternatively one can run one of several predefined queries by setting ACTION to the
+          appropriate value.
+
+          All results will be returned to the user in table format, with || as the delimeter
+          seperating multiple items within one column.
+        },
+        'Author' => [
+          'Grant Willcox', # Module
+        ],
+        'References' => [
+        ],
+        'DisclosureDate' => '2022-05-19',
+        'License' => MSF_LICENSE,
+        'Actions' => [
+          ['ENUM_ALL_OBJECTCLASS', { 'Description' => 'Dump all objects containing any objectClass field.' }],
+          ['ENUM_ALL_OBJECTCATEGORY', { 'Description' => 'Dump all objects containing any objectCategory field.' }],
+          ['ENUM_COMPUTERS', { 'Description' => 'Dump all objects containing an objectCategory of Computer.' }],
+          ['CUSTOM_QUERY', { 'Description' => 'Execute a custom LDAP query specified by LDAPQUERY.' }],
+          ['ENUM_DOMAIN_CONTROLERS', { 'Description' => 'Dump all known domain controllers.' }],
+          ['ENUM_EXCHANGE_SERVERS', { 'Description' => 'Dump info about all known Exchange servers.' }],
+          ['ENUM_EXCHANGE_RECIPIENTS', { 'Description' => 'Dump info about all known Exchange recipients.' }],
+          ['ENUM_GROUPS', { 'Description' => 'Dump info about all known groups in the LDAP environment.' }],
+          ['ENUM_ORGROLES', { 'Description' => 'Dump info about all known organizational roles in the LDAP environment.' }],
+          ['ENUM_ORGUNITS', { 'Description' => 'Dump info about all known organization units in the LDAP environment.' }],
+          ['ENUM_PEOPLE', { 'Description' => 'Dump info about all organizationalPerson objects.' }],
+          ['ENUM_USERS', { 'Description' => 'Dump info about all known users in the LDAP environement.' }]
+        ],
+        'DefaultAction' => 'ENUM_ALL_OBJECTCLASS',
+        'DefaultOptions' => {
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(389), # Set to 636 for SSL/TLS
+      OptString.new('BASE_DN', [false, 'LDAP base DN if you already have it']),
+      OptString.new('LDAPQUERY', [false, 'Query to run against the target LDAP server'])
+    ])
+  end
+
+  def perform_ldap_query(ldap, filter)
+    returned_entries = ldap.search(base: @base_dn, filter: filter)
+    if returned_entries.nil? || returned_entries.empty?
+      print_error("No results found for #{filter}. You may require additional authentication, or the information may not exist on the target.")
+      [nil, nil]
+    else
+      [filter.to_s, returned_entries]
+    end
+  end
```

Comment:
Does the filter need to be returned? If the caller needs it, they already have it.

```suggestion
  def perform_ldap_query(ldap, filter)
    returned_entries = ldap.search(base: @base_dn, filter: filter)
    if returned_entries.nil? || returned_entries.empty?
      print_error("No results found for #{filter}. You may require additional authentication, or the information may not exist on the target.")
      return nil
    end
    
    returned_entries
  end
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16598/commits/f833a5cb15b9ba4fe1b29a30cf542a2c5a30325b